src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs

```diff
@@ -190,49 +205,165 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
 
             context.RegisterOperationAction(context =>
             {
-                IInvocationOperation caseChangingInvocation = (IInvocationOperation)context.Operation;
-                IMethodSymbol caseChangingMethod = caseChangingInvocation.TargetMethod;
-
-                if (!caseChangingMethod.Equals(toLowerParameterlessMethod) &&
-                    !caseChangingMethod.Equals(toLowerInvariantParameterlessMethod) &&
-                    !caseChangingMethod.Equals(toUpperParameterlessMethod) &&
-                    !caseChangingMethod.Equals(toUpperInvariantParameterlessMethod))
```

Comment:
It seems these `toLowerParameterlessMethod, toLowerInvariantParameterlessMethod, toUpperParameterlessMethod, toUpperInvariantParameterlessMethod` symbols not used anymore, could we remove the rows populating them?
https://github.com/dotnet/roslyn-analyzers/blob/79db421c0a599950c7f0171d1f1191e0b8773c78/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs#L95-L117

Reply:
Don't we need to exit early if we don't find the symbol?

Reply:
That was actually part of the question, these symbols are not used anywhere, should we keep loading them and exit early?

Reply:
They are used here to exit early if the symbols aren't found.